pkg/apis/serving/fieldmask.go

```diff
@@ -725,7 +725,13 @@ func CapabilitiesMask(ctx context.Context, in *corev1.Capabilities) *corev1.Capa
 	// Allowed fields
 	out.Drop = in.Drop
 
-	if config.FromContextOrDefaults(ctx).Features.ContainerSpecAddCapabilities != config.Disabled {
+	if config.FromContextOrDefaults(ctx).Features.SecurePodDefaults == config.Enabled {
+		if len(in.Add) == 1 && in.Add[0] == "NET_BIND_SERVICE" {
+			out.Add = in.Add
+		} else {
+			out.Add = nil
+		}
+	} else if config.FromContextOrDefaults(ctx).Features.ContainerSpecAddCapabilities != config.Disabled {
```

Comment:
After thinking about this I think this check should happen first - and then we fall back onto the secure pod defaults check 

The reason being is that we could have both flags enable - and we would incorrectly apply the mask given the way the code is now

Reply:
So the way you explained it sounds like we actually want 2 separate statements, right? Because if both flags are enabled, `secure-pod-defaults` should override the capabilities set by the `addcapabilities` flag.

Reply:
If both flags are enabled, shouldn't `secure-pod-defaults` override `addcapabilities`? Because if we check for addCapabilities first, the user can set a capability not allowed by the restricted profile

Reply:
> If both flags are enabled, shouldn't secure-pod-defaults override addcapabilities? 

No I don't think so. If a user has the feature `addcapabilities` on then we should allow them to set all the properties.

> Because if we check for addCapabilities first, the user can set a capability not allowed by the restricted profile

That's fine - then it'll error out. If users are using `addcapaiblities` feature then they are probably not running things in a restricted profile

Reply:
Sounds good. I switched the logic to check addcapabilities first.

Reply:
Hmmm `1.26.x, kourier-tls, e2e` seems flaky. Rerunning...